test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -1,2451 +0,0 @@
-									bindings: {
-										text: {
-											source: 'core/post-meta',
-											args: { key: 'text_custom_field' },
-										},
-									},
-								},
-							},
-						},
-					],
-				} );
-				const buttonBlock = editor.canvas
-					.getByRole( 'document', {
-						name: 'Block: Button',
-						exact: true,
-					} )
-					.getByRole( 'textbox' );
-				await buttonBlock.click();
-				await expect( buttonBlock ).toHaveText(
-					'Value of the text custom field'
-				);
-
-				// Check the frontend shows the value of the custom field.
-				const previewPage = await editor.openPreviewPage();
-				const buttonDom = previewPage.locator(
-					'#button-text-binding a'
-				);
-				await expect( buttonDom ).toHaveText(
-					'Value of the text custom field'
-				);
-				await expect( buttonDom ).toHaveAttribute(
-					'href',
-					'#default-url'
-				);
-			} );
-
-			test( 'should use the value of the custom field when url is bound', async ( {
-				editor,
-			} ) => {
-				await editor.insertBlock( {
-					name: 'core/buttons',
-					innerBlocks: [
-						{
-							name: 'core/button',
-							attributes: {
-								anchor: 'button-url-binding',
-								text: 'button default text',
-								url: '#default-url',
-								metadata: {
-									bindings: {
-										url: {
-											source: 'core/post-meta',
-											args: { key: 'url_custom_field' },
-										},
-									},
-								},
-							},
-						},
-					],
-				} );
-
-				// Check the frontend shows the original value of the custom field.
-				const previewPage = await editor.openPreviewPage();
-				const buttonDom = previewPage.locator(
-					'#button-url-binding a'
-				);
-				await expect( buttonDom ).toHaveText( 'button default text' );
-				await expect( buttonDom ).toHaveAttribute(
-					'href',
-					'#url-custom-field'
-				);
-			} );
-
-			test( 'should use the values of the custom fields when text and url are bound', async ( {
-				editor,
-			} ) => {
-				await editor.insertBlock( {
-					name: 'core/buttons',
-					innerBlocks: [
-						{
-							name: 'core/button',
-							attributes: {
-								anchor: 'button-multiple-bindings',
-								text: 'button default text',
-								url: '#default-url',
-								metadata: {
-									bindings: {
-										text: {
-											source: 'core/post-meta',
-											args: { key: 'text_custom_field' },
-										},
-										url: {
-											source: 'core/post-meta',
-											args: { key: 'url_custom_field' },
-										},
-									},
-								},
-							},
-						},
-					],
-				} );
-
-				// Check the frontend uses the values of the custom fields.
-				const previewPage = await editor.openPreviewPage();
-				const buttonDom = previewPage.locator(
-					'#button-multiple-bindings a'
-				);
-				await expect( buttonDom ).toHaveText(
-					'Value of the text custom field'
-				);
-				await expect( buttonDom ).toHaveAttribute(
-					'href',
-					'#url-custom-field'
-				);
-			} );
```

Comment:
Unify testing all button attributes here for custom sources: https://github.com/WordPress/gutenberg/pull/65526/files#diff-70af8def41b67ce618e94f6b4f68044bb4540816a1a8084a1ba1eac9e0c26595R106-R140

It lacks checking the frontend.

Post meta is only checked for paragraphs.